upgrade_test.py

```diff
@@ -694,44 +698,48 @@ def test_generic_cluster_upgrade(self):  # pylint: disable=too-many-locals,too-m
 
         upgrade_sstables = self.params.get('upgrade_sstables')
 
-        if num_nodes_to_rollback > 0:
-            # Upgrade all nodes that should be rollback later
-            for i in range(num_nodes_to_rollback):
-                InfoEvent(f'Step{i + 1} - Upgrade node{i + 1}')
-                self.db_cluster.node_to_upgrade = self.db_cluster.nodes[indexes[i]]
-                self.log.info(f'Upgrade Node {self.db_cluster.node_to_upgrade.name} begins')
-                self.upgrade_node(self.db_cluster.node_to_upgrade, upgrade_sstables=upgrade_sstables)
-                self.log.info(f'Upgrade Node {self.db_cluster.node_to_upgrade.name} ended')
+        def change_node_version(node_indexes, is_upgrade=True):
```

Comment:
FYI, this refactoring is quite big, for you didn't liked some prints...

Reply:
No, I didn't create this method because of the error messages. I did it because there's duplicate code in so many places.

Reply:
I'm still not sure I like putting the logic of upgrade and rollback into the same function here
@roydahan WDYT ?

Reply:
There was a reaso why I chose not doing it.
My code is much more readable and simple.
Most of the "duplicate" code is just info events to make it easy to track in log.
without info events you simple have:

```
for i in range()
  upgrade_node
  healthcheck 
  
for node in upgraded_nodes
   rollback
   healthcheck

for node in all_nodes
   upgrade_node
   healthcheck
```

So my code is much more readable and simpler.
If I wanted to remove code lines I would embed the health_check in the upgrade or rollback procedures.

Reply:
please revert your refactor.
Including a valid "i" as index common in every computer science example.
while sometimes it's valid to call an index with a descriptive name, here it's just an helper index that make it much more readable to read the step number inside the calculations.

Reply:
@Orenef11 in summary, your code doesn't really make the code shorter, the only duplications you remove are messages to the user that should help him to debug issue based on the log.
I would claim that if he see an issue in Step3, in my case it'll be easier to understand where the issue is and in your case it will look like a black hole.

Reply:
@roydahan
In both cases, the step number does not matter because it does not exist in the code.
Examples in books and courses use these names, but this did not make them right because these names have no meaning.

I always prefer to reduce duplicate code but not at the expense of clear and clean code.
Unfortunately, in both cases, it is difficult to keep track of the logs because the step number does not appear in the blog but was built during the test.